src/DataTypes/Serializations/SerializationUUID.cpp

```diff
@@ -44,10 +45,11 @@ void SerializationUUID::serializeTextQuoted(const IColumn & column, size_t row_n
 
 void SerializationUUID::deserializeTextQuoted(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
 {
+    std::string quoted_string;
+    readQuotedStringWithSQLStyle(quoted_string, istr);
     UUID x;
-    assertChar('\'', istr);
-    readText(x, istr);
-    assertChar('\'', istr);
+    ReadBufferFromString parsed_quoted_buffer(quoted_string);
+    readText(x, parsed_quoted_buffer);
```

Comment:
Extra copy.

Reply:
I don't see it. AFAICS `ReadBufferFromString` will read directly from `quoted_string`, right?